### DISCUSSION TITLE
Battery Colours - Battery Status / Charge Colours

### DISCUSSION THREAD

**Discussion by Roving-Ronin** [slipx06/sunsynk-power-flow-card#241] [Ideas]
Hi @slipx06 

Just saw the new release with 'charge_colour:'  (gosh your making it hard to complete examples when you keep adding so many new features.. Hahahaha).

Just looking at it though and wondering would it be possible to have different colours for different SoC level % ?
Something like this (as default colours) showing 0-12% is red, 13-25% is redish yellow, 26-39% is orange etc... as per:

![full-charge-battery-battery-charging-status-indicator-bright-blue-background-3d-render_989822-859](https://github.com/slipx06/sunsynk-power-flow-card/assets/108674933/aa9d9998-9fd0-46be-8208-abf21874a04c)

battery:
  charge_colour:      (turns the text and shell of the battery to a colour, when charging.
  soc_charge_1:  red
  soc_charge_2:  'red-ish yellow'
  soc_charge_3:  orange
  soc_charge_4: 
  soc_charge_5:    .... up to 
  soc_charge_8:  light_green (or lime.. showing 100%

Have these all set as default settting colour coded in, and the above (if used) could allow setting non-default colours by users. 

If this was animated so you have it turning on the sections up to the level its currently charged at/too, then you'd also have it showing its charging and to what level.  Whilst discharging if the bars (up to its current SoC %) remain on, with the exception of the last one / current level and that flashes, it would also be graphically showing the level of SoC its down to and that its discharging.

**Comment by slipx06** [comment]
:smile:  At the moment I'm busy working on a dynamic_colour option for the battery icon when charging. It considers whether load or battery is given priority and displays percentage accordingly.  Its almost done. It might supersede your idea above or maybe this is just another display option i.e. dynamic_colour vs static_colour charge colour

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/23b4a13f-fc8d-401b-a5c2-4e25828c3d02)

**Comment by slipx06** [comment]
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/80b88d60-0be9-4460-9e52-f322b6cc9482)

**Reply by Roving-Ronin** [reply]
@slipx06 
Legend as usual ! Looks visually much better, might be better to find a slightly different colour the battery outline.. not sure but looking at it and it seems to blur with the battery level status colours (given the orange/yellow/red) used there.

v4.14.1 is it here yet.. is it here yet (said like kids on a long car trip) ;-)

**Reply by slipx06** [reply]
The battery outline colour is dynamic and will show the make up of the source power (grid or solar) feeding the battery. This can be turned off. In the example above it is 100% solar

**Reply by fruscic** [reply]
Hi, not sure mine is behaving as intended - for example, I m charging mostly from solar - say 2kw, but I get a slight 'trickle' from the grid - 20W, and the battery outline is the same colour as the grid colour. When the trickle is 0, then it goes to the solar pv colour. I expected it to be more like the load, where it is coloured proportionately?

Thanks for the integration though - it's fantastic!

**Reply by slipx06** [reply]
OK I expect there to be a couple of issues as the logic may not work for everyone.  Open to suggestion to try make it better

**Comment by slipx06** [comment]
Added to 4.15.0

**Reply by Roving-Ronin** [reply]
"priority_load_243 will determine whether the battery or the load is prioritized when generating solar power. By default the battery is prioritized."

@slipx06 just wondering, what are the responses that the card is expecting to receive for this entity?

Huawei doesn't have this my default, but it does have battery settings that specify priority 'charge battery' or 'feed (PV) to grid'. Thinking of I know what the cards expecting to see , we could do a template sensor that reformats the Huawei sensor into a sensor compatible with the card?

**Comment by slipx06** [comment]
Its a simple switch. So the value is either `on` or `off`. If undefined it assumes `off`.

**Comment by fruscic** [comment]
No problem

It works part of the time - perhaps it depends on some rounding that is
happening:

[image: image.png]

[image: image.png]

Anyway, I really use this a lot - I'm in South Africa so knowing the power
situation is critical :-)

Kind regards
Filip

**Comment by adnovea** [comment]
Nice feature. Great add-on.
THX slipx06

**Reply by Roving-Ronin** [reply]
Sorry what's it's supposed to do? Tried adding with true vs false and can't see any difference.

**Comment by adnovea** [comment]
It works fine.
 BUT if your batteries is currently discharging , because it takes the default "colour", you may see no change !

Here is my understanding :
```
  colour: '#f06292'               <- Discharging color
  charge_colour: '#88AA88'        <- Charging color
  dynamic_colour: true            <- Change color along charging source
  linear_gradient: true           <- Display gradient color on battery
```

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/44359861/dcf7237a-a8e9-42cd-91db-82fadd475c05)
_Example of charging battery screen_

**Comment by slipx06** [comment]
I guess my documentation is poor 😄 `dynamic_colour: true` is meant to show you the source of the battery charge. So if you are only using solar to charge the battery it will change the colour of the battery outline to the solar colour, or if you have specified a charge colour it will use that for the other battery elements. If you are using both grid and solar to charge the battery the percentage split should be visible  i.e. the battery outline will be a combination of grid colour and solar colour.

So in the example below the battery is charging at 38W. 19W is from solar. The orange bit and the rest from grid. The blue bit

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/23b4a13f-fc8d-401b-a5c2-4e25828c3d02)
